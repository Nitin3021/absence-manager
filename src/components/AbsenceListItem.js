import React from 'react';

const AbsenceListItem = ({ name, type, startDate, endDate, memberNote, confirmedAt, rejectedAt, admitterNote }) => (
    <div>
        <h3>
            <span>{name} </span>
            <span>{type} </span>
            <span>{startDate} </span>
            <span>{endDate} </span>
            <span>{memberNote} </span>
            <span>{confirmedAt} </span>
            <span>{rejectedAt} </span>
            <span>{admitterNote} </span>
        </h3>
    </div>
)

export default AbsenceListItem;