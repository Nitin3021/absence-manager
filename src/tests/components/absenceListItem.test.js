import React from 'react';
import { shallow } from 'enzyme';
import AbsenceListItem from '../../components/AbsenceListItem';

test('should render AbsenceListItem correctly', () => {
    const absences = [{
        "admitterId": null,
        "admitterNote": "",
        "confirmedAt": "2020-12-12T18:03:55.000+01:00",
        "createdAt": "2020-12-12T14:17:01.000+01:00",
        "crewId": 352,
        "endDate": "2021-01-13",
        "id": 2351,
        "memberNote": "",
        "rejectedAt": null,
        "startDate": "2021-01-13",
        "type": "sickness",
        "userId": 2664,
        "image": "https://loremflickr.com/300/400",
        "name": "Max",
    }]

    const wrapper = shallow(<AbsenceListItem absences={absences} />);
    expect(wrapper).toMatchSnapshot();
});