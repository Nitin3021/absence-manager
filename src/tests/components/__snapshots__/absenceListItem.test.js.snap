// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AbsenceListItem correctly 1`] = `
<div>
  <h3>
    <span>
      Max
       
    </span>
    <span>
      sickness
       
    </span>
    <span>
      2021-01-13
       
    </span>
    <span>
      2021-01-13
       
    </span>
    <span>
       
    </span>
    <span>
      2020-12-12T18:03:55.000+01:00
       
    </span>
    <span>
       
    </span>
    <span>
       
    </span>
  </h3>
</div>
`;
